lib/db/leveldb_transactions.go

```diff
@@ -149,6 +123,29 @@ func (t readWriteTransaction) updateGlobal(gk, folder, device []byte, file proto
 		t.Delete(nk)
 	}
 
+	if removedAt != 0 && insertedAt != 0 {
+		l.Debugf(`new global for "%v" after update: %v`, file.Name, fl)
+		t.Put(gk, mustMarshal(&fl))
```

Comment:
Why is the delete gone?

Reply:
All need bucket operations, including the edge case that was handled at this point before, are now handled in the block that is above (https://github.com/imsodin/syncthing/blob/6e2b90e7701b1ec86a3c4afb758d0afe9d674532/lib/db/leveldb_transactions.go#L113)